/*
 * Copyright (c) 2016-2021 Deephaven Data Labs and Patent Pending
 */

syntax = "proto3";

package io.deephaven.proto.backplane.grpc;

option java_multiple_files = true;
option optimize_for = SPEED;

import "deephaven/proto/ticket.proto";

/*
 * Allows clients to list fields that are accessible to them.
 */
service InputTableService {

    /*
     * Adds the provided tables to the specified input table. These tables must only have columns
     * (name, types, and order) which match the given input table's columns.
     */
    rpc AddTablesToInputTable(AddTableRequest) returns (AddTableResponse) {}

    /*
    * Removes the provided tables from the specified input tables. The tables indicating which rows
    * to remove are expected to only have columns that match the key columns of the input table.
    */
    rpc DeleteTablesFromInputTable(DeleteTableRequest) returns (DeleteTableResponse) {}

}

message AddTableRequest {
    Ticket input_table = 1;
    Ticket table_to_add = 2;
}

message AddTableResponse {

}

message DeleteTableRequest {
    Ticket input_table = 1;
    Ticket table_to_remove = 2;
}

message DeleteTableResponse {

}